src/irmin/irmin.mli

```diff
@@ -3619,6 +3627,9 @@ module type CONTENT_ADDRESSABLE_STORE_MAKER = functor
   val v : config -> [ `Read ] t Lwt.t
   (** [v config] is a function returning fresh store handles, with the
       configuration [config], which is provided by the backend. *)
+
+  val close : 'a t -> unit Lwt.t
+  (** [close t] frees up all the resources associated to [t]. *)
```

Comment:
It's odd that some backends will raise exceptions here if double closed, like Irmin_mem. Why not just make `close` idempotent if run multiple times, given that backends can check?